File: C++/242_valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        int len = t.length();
        unordered_map<int, int> hmap;
        
        // Create hashmap to count t
        for(int i = 0; i < len; i++){
            if(hmap.find(t[i]) == hmap.end()){
                hmap[t[i]] = 0;
            }
            else{
                hmap[t[i]]++;
            }
        }
        // Iterate through s & subtract count from hashmap t
        for(int i = 0; i < len; i++){
            if(hmap.find(s[i]) == hmap.end()){
                return false;
            }
            else{
                if(hmap[s[i]]-- < 0){
                    return false;
                }
            }
        }
        
        return true;
    }
};

File: Python/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        hmap = {}
        
        # Make hashmap to count t
        for i in t:
            hmap[i] = 1 + hmap.get(i, 0)
            
        # Iterate through s & subtract count from hashmap t
        for i in s:
            if hmap.get(i, 0) == 0:
                return False
            else:
                hmap[i] -= 1
                if hmap[i] < 0:
                    return False
            
        return True